invenio/modules/relationships/api.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+##
+## This file is part of Invenio.
+## Copyright (C) 2015 CERN.
+##
+## Invenio is free software; you can redistribute it and/or
+## modify it under the terms of the GNU General Public License as
+## published by the Free Software Foundation; either version 2 of the
+## License, or (at your option) any later version.
+##
+## Invenio is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Relationship API."""
+
+import importlib
+
+from .config import RELATIONSHIP_ENGINE, RELATIONSHIP_NODE
+
+try:
+    __engine = __import__('flask').current_app.config['RELATIONSHIP_ENGINE']
```

Comment:
There is a small problem with a `LocalProxy`. As `SmartJSON` inherits from `Node`, when the registries are run, before the application context is ready, the name is imported from this file. If I use `lazy_import` to do it, it doesn't change anything, as it still has to be imported when `SmartJSON` is defined.

To be honest, I can't see a nice solution to this problem. The one in this file is definitely wrong, although when sb will implement another engine, he will be forced to improve it.

As I mentioned before: I don't know what are the plans for integrating new `JSONAlchemy` package inside `Invenio`. As the new module is dependant on it, probably the implementation should be changed.